staging/src/k8s.io/kube-aggregator/pkg/util/dynamic_restconfig.go

```diff
@@ -0,0 +1,180 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	"k8s.io/apiserver/pkg/server/egressselector"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const workItemKey = "key"
+
+type RestConfigProvider interface {
+	GetRestConfig() (*restclient.Config, error)
+	GetTransport() (http.RoundTripper, error)
+}
+
+// DynamicRestConfigProvider provides a restconfig and transport backed by dynamically loaded cert/key pair
+type DynamicRestConfigProvider struct {
```

Comment:
this probably deserves some tests :) I'll add those after initial review validating the direction of this PR